infra/Pos/Communication/BiP.hs

```diff
@@ -11,62 +11,23 @@ module Pos.Communication.BiP
 import           Universum
 
 import qualified Data.ByteString            as BS
-import qualified Data.ByteString.Lazy       as LBS
-import qualified Data.Store                 as Store
-import qualified Network.Transport.Internal as NT (decodeWord32, encodeWord32)
 
 import           Node.Message.Class         (Serializable (..))
 import           Node.Message.Decoder       (Decoder (..))
 
-import           Pos.Binary.Class           (Bi (..), encode, label)
+import           Pos.Binary.Class           (Bi (..), serialize, decodeFull)
 
 data BiP = BiP
 
 instance  Bi t => Serializable BiP t where
```

Comment:
I would really appreciate an extra review on this bit, as the newer implementation differs quite a bit, and in particular we are not using anymore `encodeWord32` to store the length first, and I have no clue if this a dealbreaker or not.